tools/ci_build/github/apple/c/assemble_c_pod_package.py

```diff
@@ -77,14 +77,16 @@ def assemble_c_pod_package(
     # generate the podspec file from the template
     variable_substitutions = {
         "DESCRIPTION": pod_config["description"],
-        "IOS_DEPLOYMENT_TARGET": framework_info["IOS_DEPLOYMENT_TARGET"],
+         # By default, we build both "iphoneos" and "iphonesimulator" architectures, and the deployment target should be the same between these two.
+        "IOS_DEPLOYMENT_TARGET": framework_info["iphoneos"]["APPLE_DEPLOYMENT_TARGET"],
+        "MACOSX_DEPLOYMENT_TARGET": framework_info.get("macosx", {}).get("APPLE_DEPLOYMENT_TARGET", None),
```

Comment:
not sure how `re.Pattern.sub()` handles a `None` return value from the `repl` argument. that's what gets called eventually. might be safer to map to an empty string.
```suggestion
        "MACOSX_DEPLOYMENT_TARGET": framework_info.get("macosx", {}).get("APPLE_DEPLOYMENT_TARGET", ""),
```